.circleci/config.yml

```diff
@@ -270,6 +270,38 @@ jobs:
       - e2e-test:
           test_path: e2e-tests/path-prefix
 
+  e2e_tests_pnp:
+    executor: node
+    steps:
+      - <<: *attach_to_bootstrap
+      - run:
+          command: cp -r ./starters/default ./e2e-tests/gatsby-pnp
+          working_directory: ~/project
+      - run:
+          command: touch yarn.lock
+          working_directory: ~/project/e2e-tests/gatsby-pnp
+      - run: # Quick upgrade to the v2 (any version, we just need the real set version)
+          command: yarn policies set-version berry
+          working_directory: ~/project/e2e-tests/gatsby-pnp
+      - run: # Pins the Yarn version
+          command: yarn set version 2.0.0-rc.32
+          working_directory: ~/project/e2e-tests/gatsby-pnp
```

Comment:
It's a bit convoluted but since the 1.x and 2.x implementations of `set version` are different, we need to first upgrade to the 2.x before being able to pin to a specific 2.x version. I'll uniformise them in the next few weeks to use a common logic as much as possible.